File: day-09/python/part1.py
#!/usr/bin/env python3

import sys

data = [int(x.strip()) for x in open(sys.argv[1], "r").readlines()]

step = int(sys.argv[2])

for i in range(step, len(data)):
    found = False
    for j in range(i-step, i):
        for k in range(i-step, i):
            if k!=j and data[i] == data[k] + data[j]:
                found = True
    if not found:
        print(i, data[i])

File: day-09/nim/part1.nim
from os import paramStr
import sugar, strutils

let 
  data = collect(newSeq):
    for x in lines(paramStr(1)):
      parseInt(x.strip())
  step = parseInt(paramStr(2))

for i in step ..< data.len:
  var found = false
  for j in (i - step) ..< i:
    for k in (i - step) ..< i:
      if k!=j and data[i] == data[k] + data[j]:
        found = true
  if not found:
    echo data[i]
